pyro/util.py

```diff
@@ -324,6 +324,30 @@ def check_if_enumerated(guide_trace):
             'If you want to enumerate sites, you need to use TraceEnum_ELBO instead.']))
 
 
+@contextmanager
+def ignore_jit_warnings(filter=None):
+    """
+    Ignore JIT tracer warnings with messages that match `filter`. If
+    `filter` is not specified all tracer warnings are ignored.
+
+    :param filter: A list containing either warning message (str),
+        or tuple consisting of (warning message (str), Warning class).
+    """
+    with warnings.catch_warnings():
+        if filter is None:
+            warnings.filterwarnings("ignore",
+                                    category=torch.jit.TracerWarning)
+        else:
+            category = torch.jit.TracerWarning
```

Comment:
Move this down one line, otherwise args to `ignore_jit_warnings` are order-dependent:
```py
for msg in filter:
    category = torch.jit.TracerWarning
    if isinstance(msg, tuple):
        msg, category = msg
    ...
```

Reply:
My bad; will fix.